### PR TITLE
WIP: parametrize giant frame flex op tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_arithmetic.py [hunk 1]
```diff
@@ -138,3 +143,252 @@ def test_ops_frame_period(self):
...
         tm.assert_frame_equal(df2 - df, exp)
         tm.assert_frame_equal(df - df2, -exp)
+
+
+class TestFrameFlexArithmetic(object):
+    """Tests for DataFrame flex operations, mostly focused on alignment
+    and broadcasting.
+    """
+    def test_df_flex_add_int(self):
+        frame = pd.DataFrame(tm.getSeriesData())
```
Comment (gfyoung) [inline pandas/tests/frame/test_arithmetic.py]: I wonder if we should make `pd.DataFrame(tm.getSeriesData())` its own `tm` method given how often you use it.

File: pandas/tests/frame/test_arithmetic.py [hunk 2]
```diff
@@ -138,3 +143,252 @@ def test_ops_frame_period(self):
...
+
+        result = frame.add(frame[:0])
+        expected = frame * np.nan
+        tm.assert_frame_equal(result, expected)
+
+        result = frame[:0].add(frame)
+        tm.assert_frame_equal(result, frame * np.nan)
+
+    def test_df_flex_add_fill_value_raises(self):
+        # corner case NotImplementeError
```
Comment (gfyoung) [inline pandas/tests/frame/test_arithmetic.py]: `NotImplemente` --> `NotImplementedError`

File: pandas/tests/frame/test_arithmetic.py [hunk 3]
```diff
@@ -138,3 +143,252 @@ def test_ops_frame_period(self):
...
+
+        dunder_func = getattr(operator, opname)
+        rfunc = lambda x, y: dunder_func(y, x)
+
+        result = getattr(frame, 'r' + opname)(2 * frame)
+        expected = rfunc(frame, 2 * frame)
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize('opname', ['add', 'sub', 'mul'])
+    def test_df_flex_needs_better_name(self, opname):
```
Comment (gfyoung) [inline pandas/tests/frame/test_arithmetic.py]: Looks like the test concerns mixed types.

File: pandas/tests/frame/test_arithmetic.py [hunk 4]
```diff
@@ -138,3 +143,252 @@ def test_ops_frame_period(self):
...
+        try:
+            dunder_func = getattr(operator, alias)
+
+            # vs mix float
+            result = getattr(mixed_float, opname)(2 * mixed_float)
+            expected = dunder_func(mixed_float, 2 * mixed_float)
+            tm.assert_frame_equal(result, expected)
+            _check_mixed_float(result, dtype={'C': None})
+
+        except:
```
Comment (gfyoung) [inline pandas/tests/frame/test_arithmetic.py]: Bare except...boo...can we make this more specific?  Even `except Exception` is better actually.

File: pandas/tests/frame/test_arithmetic.py [hunk 5]
```diff
@@ -138,3 +143,252 @@ def test_ops_frame_period(self):
...
+            import pandas.io.formats.printing as printing
+            printing.pprint_thing("Failing operation %r" % opname)
+            raise
+
+    def test_df_arith_flex(self):
+        pass
```
Comment (gfyoung) [inline pandas/tests/frame/test_arithmetic.py]: What happened here?

Comment (jbrockmendel) [reply pandas/tests/frame/test_arithmetic.py]: These tests are being split from (see the diff of `test_operators`) `test_arith_flex_frame`, which has these lines commented-out in the status quo.  Retained them on the "they're here for a reason, I just don't know it yet" principle.

**Comment by jbrockmendel** [comment]
Closing for now to clear out the queue, will circle back.

**Pull request closed without merging**